### PR TITLE
fix: [CDS-82973]: allow clearing value for Downsize Instance Unit field

### PR DESCRIPTION
**Pull request by chetanvekariya-harness**
### Summary
- Used Dropdown field instead of FormInput.Select field to allow clearing value for Downsize Instance Unit field. Tried using normal Select, MultiTypeInput with allowableType as empty array, but these solutions did not work as value was not getting cleared even if click happens on clear button. There seems to be an issue with internal state of Select component in uicore.
- Added test cases to cover use case of clearing values for optional fields

#### Screenshots
Screenshot 2023-11-06 at 4 48 23 PM

https://github.com/harness/harness-core-ui/assets/81295223/2af3a785-936b-4ca2-9d02-ae2b514ddab3

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by chetanvekariya-harness** [comment]
trigger messagecheck

**Review by litvinovstas — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/AmazonECS/ECSUpgradeContainerStep/ECSUpgradeContainerStep.tsx [hunk 1]
```diff
@@ -90,17 +90,27 @@ export class ECSUpgradeContainerStep extends PipelineStep<ECSUpgradeContainerSte
...
-        onChange={onChange}
         stepViewType={stepViewType}
         ref={formikRef}
         readonly={readonly}
       />
     )
   }
 
+  processFormData(values: ECSUpgradeContainerStepElementConfig): ECSUpgradeContainerStepElementConfig {
+    if ((values.spec.downsizeOldServiceInstanceCount as string)?.length === 0) {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/AmazonECS/ECSUpgradeContainerStep/ECSUpgradeContainerStep.tsx:105]: As `downsizeOldServiceInstanceCount?: number | string`, Do we still need to check for length?

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/AmazonECS/ECSUpgradeContainerStep/ECSUpgradeContainerStep.tsx (hunk 1 above)
Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/AmazonECS/ECSUpgradeContainerStep/ECSUpgradeContainerStep.tsx:105]: Yes, clearing the value makes it empty string, so need to check for length

Comment (chetanvekariya-harness) [reply src/modules/75-cd/components/PipelineSteps/AmazonECS/ECSUpgradeContainerStep/ECSUpgradeContainerStep.tsx]: Used toString to convert number into string before using length.

**Comment by chetanvekariya-harness** [comment]
retrigger eslint

**Review by rameshx — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**